hnn_core/tests/test_viz.py

```diff
@@ -208,4 +217,108 @@ def test_dipole_visualization():
+        _ = NetworkPlotter(net, xlim='blah')
+    with pytest.raises(TypeError, match='ylim must be'):
+        _ = NetworkPlotter(net, ylim='blah')
+    with pytest.raises(TypeError, match='zlim must be'):
+        _ = NetworkPlotter(net, zlim='blah')
+    with pytest.raises(TypeError, match='elev must be'):
+        _ = NetworkPlotter(net, elev='blah')
+    with pytest.raises(TypeError, match='azim must be'):
+        _ = NetworkPlotter(net, azim='blah')
+    with pytest.raises(TypeError, match='vmin must be'):
+        _ = NetworkPlotter(net, vmin='blah')
+    with pytest.raises(TypeError, match='vmax must be'):
+        _ = NetworkPlotter(net, vmax='blah')
+    with pytest.raises(TypeError, match='trial_idx must be'):
+        _ = NetworkPlotter(net, trial_idx=1.0)
+    with pytest.raises(TypeError, match='time_idx must be'):
+        _ = NetworkPlotter(net, time_idx=1.0)
+
+    net = jones_2009_model(params)
+    net_plot = NetworkPlotter(net)
+
+    assert net_plot.vsec_array.shape == (159, 1)
+    assert net_plot.color_array.shape == (159, 1, 4)
+    assert net_plot._vsec_recorded is False
+
+    # Errors if vsec isn't recorded
+    with pytest.raises(RuntimeError, match='Network must be simulated'):
+        net_plot.export_movie('demo.gif', dpi=200)
+
+    # Errors if vsec isn't recorded with record_vsec='all'
+    _ = simulate_dipole(net, dt=0.5, tstop=10, record_vsec='soma')
+    net_plot = NetworkPlotter(net)
+
+    assert net_plot.vsec_array.shape == (159, 1)
+    assert net_plot.color_array.shape == (159, 1, 4)
+    assert net_plot._vsec_recorded is False
+
+    with pytest.raises(RuntimeError, match='Network must be simulated'):
+        net_plot.export_movie('demo.gif', dpi=200)
+
+    # Simulate with record_vsec='all' to test voltage plotting
+    net = jones_2009_model(params)
+    _ = simulate_dipole(net, dt=0.5, tstop=10, n_trials=2,
+                        record_vsec='all')
+    net_plot = NetworkPlotter(net)
+
+    assert net_plot.vsec_array.shape == (159, 21)
+    assert net_plot.color_array.shape == (159, 21, 4)
+    assert net_plot._vsec_recorded is True
+
+    # Type check errors
+    with pytest.raises(TypeError, match='xlim must be'):
+        net_plot.xlim = 'blah'
+    with pytest.raises(TypeError, match='ylim must be'):
+        net_plot.ylim = 'blah'
+    with pytest.raises(TypeError, match='zlim must be'):
+        net_plot.zlim = 'blah'
+    with pytest.raises(TypeError, match='elev must be'):
+        net_plot.elev = 'blah'
+    with pytest.raises(TypeError, match='azim must be'):
+        net_plot.azim = 'blah'
+    with pytest.raises(TypeError, match='vmin must be'):
+        net_plot.vmin = 'blah'
+    with pytest.raises(TypeError, match='vmax must be'):
+        net_plot.vmax = 'blah'
+    with pytest.raises(TypeError, match='trial_idx must be'):
+        net_plot.trial_idx = 1.0
+    with pytest.raises(TypeError, match='time_idx must be'):
+        net_plot.time_idx = 1.0
```

Comment:
why is this repeated?

Reply:
It is redundant, but I figure there should be separate tests for setting parameters before or after initialization

Reply:
This is a nitpick, but could we put these tests in `for` loop to conserve space + enhance readability?